### ISSUE TITLE
Don't allow creation of a directory channel if no directory facets exist

### ISSUE THREAD

**Issue by danchamp** [localgovdrupal/localgov_directories#106]
If there are no directory facets, the user needs a warning that they need to create at least one directory facet before creating a directory channel.

<img width="1227" alt="image" src="https://user-images.githubusercontent.com/261421/120331764-052c7b80-c2e6-11eb-8291-2f25280fc8c6.png">

**Comment by ekes** [comment]
I'd go the other way. It should be possible to make directories without the 'Facets' (by which I primarily mean the content facets that we created). Some councils will have directories that only have controlled vocabularies (probably Taxonomy Terms) and no editorial on-the-fly content facets (I seem to remember we had to allow this functionality for Lambeth).

**Comment by danchamp** [comment]
That makes sense @ekes

**Comment by andybroomfield** [comment]
I don't think we should force facets to be created, there are cases when a simple directory would be needed and it won't have any facets.

**Comment by danchamp** [comment]
> I don't think we should force facets to be created, there are cases when a simple directory would be needed and it won't have any facets.

I agree, but had assumed that facets are currently required for a reason. If there's no documented reason for facets being required I'll close this issue and open a new issue for removing the requirement.

**Comment by ekes** [comment]
I think this is a classic use different councils different use cases thing.
It's a two part question: out-of-the-box what should the set up be; what should be possible to configure.
It is certainly possible to make a directory that doesn't need facets; but you need to set up the content type now.
So are the defaults, requiring a 'content facet', as the default too restrictive? Or is this correct and (a) there should be better UX for  content administrators about how you set up a directory; and (b) there should be better guidance for site builders how to make directories without facets and/or with config taxonomy term vocabularies. If content facets aren't required should there be better UX for content administrators to guide them to making the (as if there aren't config taxonomy vocabularies) it is likely that they will be desired and won't be searched for unless there is a message?

**Comment by ekes** [comment]
> I think this is a classic use different councils different use cases thing.
> It's a two part question: out-of-the-box what should the set up be; what should be possible to configure.
> It is certainly possible to make a directory that doesn't need facets; but you need to set up the content type now.
> So are the defaults, requiring a 'content facet', as the default too restrictive? Or is this correct and (a) there should be better UX for content administrators about how you set up a directory; and (b) there should be better guidance for site builders how to make directories without facets and/or with config taxonomy term vocabularies. If content facets aren't required should there be better UX for content administrators to guide them to making the (as if there aren't config taxonomy vocabularies) it is likely that they will be desired and won't be searched for unless there is a message?

And to be clear. I'm 100% agnostic on which way it goes. It should be decided which way it goes, and then the work done to make doing the various options easier and more obvious.

**Comment by danchamp** [comment]
> And to be clear. I'm 100% agnostic on which way it goes. It should be decided which way it goes, and then the work done to make doing the various options easier and more obvious.

I agree with all the above. On the basis that facets *are* currently required, and echoing #69, a logical first step would be a warning to the user if no facets exists when they start creating a new channel.

Ultimately I don't think facets being required is helpful - next time I come to create a new channel which needs new facets, I'll be forced to either select a facet I don't want, and go back to remove it once I've created the new facet I do want, or abandon the new channel and go sort the facet first.

I.e. a warning can't possibly tell me if the facet I want in my channel exists or not. :-)
